lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -124,150 +124,128 @@ def run_module():
+                        vnic_uuid = boot_order_item.vnic_uuid
+                        vnic_name = boot_order_item.name
+
+            else:
+                # TODO - his will eventually need to be modified so that any
+                # boot order is possible for any extra NICs
+
+                vnic_uuid = str(uuid4())
+                vnic_boot_order = len(boot_order) + i
+
+                vnic_payload = tacp.ApiAddVnicPayload(
+                    automatic_mac_address=automatic_mac_address,
+                    name=vnic_name,
+                    network_uuid=network_uuid,
+                    boot_order=vnic_boot_order,
+                    mac_address=mac_address
+                )
+                vnic_payloads.append(vnic_payload)
+
+            network_payload = tacp.ApiCreateOrEditApplicationNetworkOptionsPayload(
                 name=vnic_name,
+                automatic_mac_assignment=automatic_mac_address,
                 network_uuid=network_uuid,
-                boot_order=vnic_boot_order,
+                vnic_uuid=vnic_uuid,
                 mac_address=mac_address
             )
-            vnic_payloads.append(vnic_payload)
-
-        network_payload = tacp.ApiCreateOrEditApplicationNetworkOptionsPayload(
-            name=vnic_name,
-            automatic_mac_assignment=automatic_mac_address,
-            network_uuid=network_uuid,
-            vnic_uuid=vnic_uuid,
-            mac_address=mac_address
-        )
-        network_payloads.append(network_payload)
-
-    instance_params['boot_order'] = boot_order
-    instance_params['networks'] = network_payloads
-    instance_params['vnics'] = vnic_payloads
-    instance_params['vcpus'] = module.params['vcpu_cores']
-    instance_params['memory'] = convert_memory_abbreviation_to_bytes(
-        module.params['memory'])
-    instance_params['vm_mode'] = module.params['vm_mode'].capitalize()
-    instance_params['vtx_enabled'] = module.params['vtx_enabled']
-    instance_params['auto_recovery_enabled'] = module.params['auto_recovery_enabled']
-    instance_params['description'] = module.params['description']
-
-    def create_instance(instance_params):
-        api_instance = tacp.ApplicationsApi(tacp.ApiClient(configuration))
+            network_payloads.append(network_payload)
+
+            instance_params['boot_order'] = boot_order
+            instance_params['networks'] = network_payloads
+            instance_params['vnics'] = vnic_payloads
+            instance_params['vcpus'] = module.params['vcpu_cores']
+            instance_params['memory'] = tacp_utils.convert_memory_abbreviation_to_bytes(
+                module.params['memory'])
+            instance_params['vm_mode'] = module.params['vm_mode'].capitalize()
+            instance_params['vtx_enabled'] = module.params['vtx_enabled']
+            instance_params['auto_recovery_enabled'] = module.params['auto_recovery_enabled']
+            instance_params['description'] = module.params['description']
```

Comment:
An idea for future refactoring: we might be better with using schematics or some sort of data model with validation to ease things up a little bit

Reply:
I'm not familiar with schematics, but I like the idea of better validation for sure :)